src/placement/pl_map.c

```diff
@@ -607,7 +599,7 @@ pl_select_leader(daos_obj_id_t oid, uint32_t grp_idx, uint32_t grp_size,
 		return shard->po_shard;
 	}
 
-	/* XXX: The shards within [start, start + replicas) will search from
+	/* XXX: The shards within [start, start + grp_size) will search from
```

Comment:
Not sure I follow here, I thought leader selection should based on "fixed" group_size from object_class. instead of group size in layout, which may include extending shards.

Reply:
i.e. leader should keep same for inflight I/O during the pool extending.

Reply:
leader keeps same on same pool map version, when pool map refreshed (group_size possibly change in pool extending case) the leader possibly be changed. But this is same as before right?

Reply:
@wangdi1 refreshed as you suggested, please check it again, thx!